<!DOCTYPE html>
<!-- saved from url=(0084)http://software.broadinstitute.org/cancer/software/genepattern/modules/docs/DESeq2/1 -->
<html class=""><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  <title>DESeq2 (v1.0)</title>
  <link href="./application.css" media="all" rel="stylesheet">
  <script src="./application.js"></script><style>.cke{visibility:hidden;}</style><style type="text/css"></style>
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta content="authenticity_token" name="csrf-param">
<meta content="6R242e9gOwhB3aUx1DZmAI3WOTkEOG5maGP7nsyYfD4=" name="csrf-token">
  <link href="http://software.broadinstitute.org/cancer/software/genepattern/assets/favicon-2382cf779a3f7287bc54add7e6f6b08d.ico" rel="shortcut icon" type="image/vnd.microsoft.icon">

<style type="text/css">.fancybox-margin{margin-right:15px;}</style></head>
<body>

	<div class="gp-content-header fluid">
	<div class="container">
		<h1>DESeq2 (v1.0) <a style="float: right" href="https://www.genepattern.org"><img alt="GP Logo" src="gplogo.png" /></a></h1>
	</div>
</div>
<div class="container">
	<div class="row">
		<div class="col-md-12">
			<p style="font-size: 1.2em; margin-top: 20px;">A Differential Expression workflow for RNA-Seq data for the GenePattern platform, based on the DESeq2 package from Bioconductor </p>
			<div class="row">
				<div class="col-sm-4">
					<p><strong>Author: </strong>David Eby;Broad Institute</p>
				</div>
				<div class="col-sm-4">
					<p><strong>Contact: </strong></p><p>gp-help@broadinstitute.org</p>
<p></p>
				</div>
				<div class="col-sm-4">
					<p><strong>Algorithm Version: </strong>DESeq2_1.10.1 </p>
				</div>
			</div>

			<div class="row">
				<div class="col-md-12">
					<h2>Introduction</h2>

<p dir="ltr">One common task in bioinformatics is the analysis of high-throughput RNA-seq data for the purpose of finding genes which are differentially expressed across groups of samples or phenotypes. This can be accomplished by quantification and evaluation of the read count -- the number of transcripts that map to each genomic region -- which is an approximate measurement of the abundance of the target transcript [1]. Typically, read counts are mapped and attributed to genes, allowing a user to evaluate the transcription at a gene level. To determine differential expression, read counts can be compared between different classes, which are groupings of samples based on, e.g. differing phenotypes, barcodes, or transcription factor binding sites.</p>

<h2>Algorithm</h2>

<p>Although many methods exist for analyzing RNA-seq read counts, several challenges exist. Due to the nature of RNA sequencing, the resulting data can have small replicate numbers, a large dynamic range, or contain many outliers. Therefore, analysis of RNA-seq data must take these issues into account.</p>

<p dir="ltr">DESeq2 is a popular algorithm for analyzing RNA-seq data [2], which estimates the variance-mean depending in high-throughput count data, and determines differential expression based on a negative binomial distribution [3]. DESeq2 improves upon the previously published DESeq algorithm, by improving stability and interpretability of expression estimates.</p>

<p dir="ltr">The GenePattern DESeq2 module takes RNA-Seq raw count data as an input, in the GCT file format. These raw count values can be generated by HTSeq-Count [4], which determines un-normalized count values from aligned sequencing reads and a list of genomic features (e.g. genes or exons). The HTSeq-Count tool is not currently&nbsp;available on GenePattern. If you have data from HTSeq-counts, the&nbsp;GenePattern MergeHTSeqCountx module will merge multiple samples together into one GCT file, which can then be passed to DESeq2. Note that DESeq2 will not accept normalized RPKM or FPKM values, only raw count data.</p>

<p dir="ltr">In addition to the raw count data in GCT format, the DESeq2 module requires a class file in the CLS format, which categorizes samples based on, e.g. phenotype, treatment, cell type, etc. In addition, DESeq2 accepts a confounding class file, which re-categorizes samples according to a secondary class, allowing for two-factor analysis. DESeq2 completes differential expression analysis based on the provided class files. For more than two classes, the module offers the option of comparing either in a “one-versus-all” approach, or “all-pairs” approach.</p>

<p>In the following example, we have a two-factor analysis grouped by two sets of classes: (1) cell type (acute myeloid leukemia vs. multiple myeloma); and (2) treatment type (drug-treated vs. untreated). Since there are two variants in each class, this gives us a total of four classes.</p>

<p><span id="docs-internal-guid-5c9e47b6-2c2e-ed6d-3144-30a64454d394"><span style="font-size: 14.6667px; font-family: Arial; color: rgb(0, 0, 0); background-color: transparent; vertical-align: baseline; white-space: pre-wrap;"><img height="181" src="./example_classes.png" style="border-width: initial; border-style: none; transform: rotate(0rad);" width="361"></span></span></p>

<p dir="ltr">If we wanted to complete a single-factor analysis, e.g. just by treatment type, we would provide only one CLS file, and would not need to specify whether we want a “one-versus-all” or “all-pairs” analysis, since there are only two classes (drug-treated vs. untreated). However, since we have elected a two-factor approach, we provide a primary CLS file (cell type) and a confounding CLS file (treatment type), and must choose a type of analysis, resulting in the following comparisons:</p>

<ul>
	<li dir="ltr">
	<p dir="ltr"><strong>one-versus-all</strong>: &nbsp;“A vs. (B and C and D)”, “B vs. (A and C and D)”, “C vs. (A and B and D)”, and “D vs. (A and B and C)”</p>
	</li>
	<li dir="ltr">
	<p dir="ltr"><strong>all-pairs</strong>: “A vs. B”, “A vs. C”, “A vs. D”, “B vs. C”, “B vs. D”, “C vs. D”</p>
	</li>
</ul>

<p>&nbsp;</p>

<h2>References</h2>

<ol>
	<li>Love MI, Huber W and Anders S (2014). “Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2.”&nbsp;<em>Genome Biology</em>,&nbsp;15, pp. 550.&nbsp;<a href="http://doi.org/10.1186/s13059-014-0550-8">doi: 10.1186/s13059-014-0550-8</a>.</li>
	<li>"DESeq2"&nbsp;<em>Bioconductor,&nbsp;</em>Bioconductor version: Release (3.4),&nbsp;<a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html" target="_blank">https://bioconductor.org/packages/release/bioc/html/DESeq2.html</a>, Accessed 03 Nov., 2016.</li>
	<li>David Bridges, "Why do we use the negative binomial distribution for analysing RNAseq data?",&nbsp;<em>Bridges Lab,&nbsp;</em>30 Aug.&nbsp;2015,&nbsp;<a href="http://bridgeslab.sph.umich.edu/posts/why-do-we-use-the-negative-binomial-distribution-for-rnaseq" target="_blank">http://bridgeslab.sph.umich.edu/posts/why-do-we-use-the-negative-binomial-distribution-for-rnaseq</a>, Accessed 03 Nov. 2016</li>
	<li>"Counting reads in features with&nbsp;htseq-count<tt class="docutils literal" style="background-color: transparent; padding: 0px 1px; font-size: 0.95em;"><span class="pre">"&nbsp;</span></tt><em>HTSeq 0.6.1p2 documentation,</em><tt class="docutils literal" style="background-color: transparent; padding: 0px 1px; font-size: 0.95em;"><span class="pre"><em>&nbsp;</em></span></tt><a href="http://www-huber.embl.de/users/anders/HTSeq/doc/count.html" target="_blank">http://www-huber.embl.de/users/anders/HTSeq/doc/count.html</a>, Accessed 03 Nov. 2016</li>
	<li>"Wald test." &nbsp;<em>Wikipedia,&nbsp;</em><a href="https://en.wikipedia.org/wiki/Wald_test" target="_blank">https://en.wikipedia.org/wiki/Wald_test</a>, Accessed 03 Nov. 2016</li>
	<li>Michael Love, Simon Anders, Vladislav Kim&nbsp;and Wolfgang Huber. "RNA-seq workflow: gene-level exploratory analysis and differential expression" <em>Bioconductor</em>, <a href="http://www.bioconductor.org/help/workflows/rnaseqGene/#plotting-results" target="_blank">http://www.bioconductor.org/help/workflows/rnaseqGene/#plotting-results</a>. Accessed 03 Nov. 2016&nbsp;</li>
</ol>

				</div>
			</div>
			<div class="row">
				<div class="col-md-12">
					<h2>Parameters</h2>

<table class="table table-striped">
	<thead>
		<tr>
			<th>Name</th>
			<th>Description</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<td>input file&nbsp;<span style="color:red;">*</span></td>
			<td>A GCT file containing raw RNA-Seq counts, such as is produced by MergeHTSeqCounts</td>
		</tr>
		<tr>
			<td>cls file&nbsp;<span style="color:red;">*</span></td>
			<td>A categorical CLS file specifying the phenotype classes for the samples in the GCT file. This should contain exactly two classes with the control specified first.</td>
		</tr>
		<tr>
			<td>confounding variable cls file&nbsp;</td>
			<td>A categorical CLS file specifying an additional confounding variable, mapped to the input file samples. Use this for a two-factor comparison.</td>
		</tr>
		<tr>
			<td>output file base&nbsp;<span style="color:red;">*</span></td>
			<td>The base name of the output file(s). File extensions will be added automatically.</td>
		</tr>
		<tr>
			<td>qc plot format&nbsp;<span style="color:red;">*</span></td>
			<td>Choose a file format for the QC plots (or skip them completely)</td>
		</tr>
		<tr>
			<td>fdr threshold&nbsp;<span style="color:red;">*</span></td>
			<td>FDR threshold to use in filtering before creating the reports of top up-regulated and down-regulated genes</td>
		</tr>
		<tr>
			<td>top N count&nbsp;<span style="color:red;">*</span></td>
			<td>Max gene count to use when creating the reports of top up-regulated and down-regulated genes</td>
		</tr>
		<tr>
			<td>random seed&nbsp;<span style="color:red;">*</span></td>
			<td>Seed to use for randomization</td>
		</tr>
	</tbody>
</table>

<p><span style="color: red;">*</span> - required</p>

				</div>
			</div>
			<div class="row">
				<div class="col-md-12">
					<h2>Input Files</h2>

<ol>
	<li><strong>input file</strong><br>
	The GenePattern DESeq2 module takes RNA-Seq raw count data as an input, in the GCT file format. These raw count values can be generated by HTSeq-Count, which determines un-normalized count values from aligned sequencing reads and a list of genomic features (e.g. genes or exons). The HTSeq-Count tool is not yet available on GenePattern. If you have .count data from HTSeq-count, run outside of GenePattern, the GenePattern MergeHTSeqCounts module will merge multiple samples together into one GCT file, which can then be passed to DESeq2. Note that DESeq2 will not accept normalized RPKM or FPKM values, only raw count data.</li>
	<li><strong>cls file</strong><br>
	In addition to the raw count data in GCT format, the DESeq2 module requires a class file in the CLS format, which categorizes samples based on, e.g. phenotype, treatment, cell type, etc.</li>
	<li><strong>confounding variable cls file</strong><br>
	In addition, DESeq2 accepts a confounding class file, which re-categorizes samples according to a secondary class, allowing for two-factor analysis.</li>
</ol>

<h2>Output Files</h2>

<p>DESeq2 will create a several output files<em> for each comparison</em>:</p>

<p style="margin-left: 40px;">The results of the DESeq2 analysis. The results are automatically ordered by the “padj” variable, which is the Benjamini-Hochberg adjusted p-value for multiple hypothesis testing. The results columns are as follows:</p>

<ol>
	<li><strong>*_results_report.TXT</strong>

	<ul dir="ltr">
		<li style="margin-left: 80px;"><strong>id</strong>: the feature identification value, e.g. gene name, Ensembl ID, barcode, etc.</li>
		<li style="margin-left: 80px;"><strong>baseMean</strong>: intermediate mean of normalized counts for all samples. This considers all samples in the dataset and determines the average normalized count value, dividing by size factors.</li>
		<li style="margin-left: 80px;"><strong>log2FoldChange</strong>: log2 fold change when comparing two classes. This estimates effect size. It says how much a gene’s expression seems to have changed due to, e.g. treatment (class 1), when compared to untreated samples (class 2).</li>
		<li style="margin-left: 80px;"><strong>lfcSE</strong>: standard error estimate for the log2 fold change estimate.</li>
		<li style="margin-left: 80px;"><strong>stat</strong>: Wald statistic [5]; this is the Wald test statistic result for the log2 fold change when comparing classes.</li>
		<li style="margin-left: 80px;"><strong>pvalue</strong>: Wald test p-value</li>
		<li style="margin-left: 80px;"><strong>padj</strong>: Benjamini-Hochberg adjusted p-value</li>
	</ul>
	</li>
	<li><strong>*_mean_values_by_class.TXT:</strong><br>
	The per-row mean of values organized by class, with the same ordering as in the *_results_report.txt file. This file contains the feature ID and the per-row mean for each class in individual columns.</li>
	<li><strong>*_topN_upregulated_genes_report.TXT</strong>:<br>
	Filters the <em>*_results_report.TXT</em> file into the top <strong>N </strong>up-regulated features, based on the log2 fold change value after filtering<em> padj </em>&lt; 0.1. The <em>padj</em> filtering threshold and the number of reported genes are adjustable.</li>
	<li><strong>*_topN_downregulated_genes_report.TXT:</strong><br>
	Filters the <em>*_results_report.TXT</em> file into the top<strong> N</strong> down-regulated features, based on the log2 fold change value after filtering <em>padj </em>&lt; 0.1. The <em>padj</em> filtering threshold and the number of reported genes are adjustable.</li>
	<li><strong>*.normalized_counts.TXT and *.normalized_counts.GCT:</strong><br>
	The normalized counts table, both as a text report and in GCT file format such that it can be supplied to GSEA or other GenePattern modules.</li>
</ol>

<p><strong>Optional Output Files:</strong></p>

<ol>
	<li><strong>*.MAplot.png:</strong><br>
	An MA plot of log2 fold change compared to the mean expression values for each feature. Red points indicate features for which the log2 fold change is significantly higher than 1 (up-regulated) or less than -1 (down-regulated) when comparing classes.</li>
	<li><strong>*DispEst.png</strong><br>
	A Disperson Estimates plot, which shows the dispersion values for each gene plotted against the mean of normalized counts.</li>
</ol>

<p dir="ltr"><em>Note that, to improve performance and resource usage, the DESeq2 module pre-filters the dataset by removing any rows where the total read count is either 0 or 1; e.g. the feature has no expression value in any available class. Thus, the total rows in the results files may not match the number of rows in the input files.</em></p>

<div>&nbsp;</div>

<h2>Example Data</h2>

<p>Input files can be found here :&nbsp;<a href="https://github.com/genepattern/DESeq2/tree/develop/gpunit/Input" target="_blank">https://github.com/genepattern/DESeq2/tree/develop/gpunit/Input</a></p>

<p>Example output can be found here:&nbsp;<a href="https://github.com/genepattern/DESeq2/tree/develop/gpunit/Output" target="_blank">https://github.com/genepattern/DESeq2/tree/develop/gpunit/Output</a></p>

<p><em>Note that the output was created by running with the example data, and all defaults, except that the module was configured to output pdf plots.</em></p>

<h2>Requirements</h2>

<p>This module requires R version 3.2.5 to be installed and for the server to be able to resolve the substitution value &lt;R3.2_Rscript&gt;, found on the command line.</p>

<p>Please see our Adminstrators guide for more information, or feel free to <a href="http://software.broadinstitute.org/cancer/software/genepattern/contact" target="_blank">contact us</a> directly.</p>

				</div>
			</div>

			<h2>Platform Dependencies</h2>

			<div class="row">
				<div class="col-md-3">
					<p><strong>Task Type:</strong><br>
					Gene List Selection</p>
				</div>
				<div class="col-md-3">
					<p><strong>CPU Type:</strong><br>
					any</p>
				</div>
				<div class="col-md-3">
					<p><strong>Operating System:</strong><br>
					any</p>
				</div>
				<div class="col-md-3">
					<p><strong>Language:</strong><br>
					R3.2.5</p>
				</div>
			</div>

			<h2>Version Comments</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>Version</th>
						<th>Release Date</th>
						<th>Description</th>
					</tr>
				</thead>
				<tbody>
						<tr>
							<td>1.0</td>
							<td>2019-10-16</td>
							<td>Updated to output a counts table file</td>
						</tr>
						<tr>
							<td>0.2</td>
							<td>2016-11-03</td>
							<td>Initial Beta Release</td>
						</tr>
				</tbody>
			</table>
		</div>
	</div>
</div>


	<footer class="container fluid gp-footer">
	<div class="fluid text-center">©2006-2019 Regents of the University of California &amp; Broad Institute</div> 
</footer>


</body></html>